build/helper/documentation_snippets.py

```diff
@@ -105,6 +113,45 @@
 '''
 
 
+def close_function_def_for_doc(functions):
+    # This is very specific to session based APIs. We look for a 'close' function and if we find it,
+    # We will copy that and modify it to be what we need for documentation
+    if 'close' in functions:
+        import copy
+        function_def = copy.deepcopy(functions['close'])
+        if 'documentation' not in function_def:
+            function_def['documentation'] = {}
+        if 'description' not in function_def['documentation']:
+            function_def['documentation']['description'] = default_close_function_doc
+        if 'note' not in function_def['documentation']:
+            function_def['documentation']['note'] = []
+        if type(function_def['documentation']['note']) is not list:
+            function_def['documentation']['note'] = [function_def['documentation']['note']]
+        function_def['documentation']['note'].append(close_function_note)
+        function_def['python_name'] = 'close'
+    else:
```

Comment:
Maybe we should consider breaking the build if we don't have documentation for `close` (or `initiate`).
I assume having the `else` case is nice right now because `nidigital` doesn't have documentation. If that's the reason for having this, we should consider adding documentation via add-on just for these two functions.

Reply:
Done. Assert when close or initiate function doesn't exist

Also found that `'close_function'` in config metadata was incorrect. Fixed here and internally.